### ISSUE TITLE
Checkout block: Add warning notice for shipping section within the Order summary

### ISSUE THREAD

**Issue by nielslange** [woothemes/woocommerce#42422]
> [!CAUTION]
> This issue is currently blocked as the outcome need to be defined. See https://github.com/woocommerce/woocommerce/issues/42422#issuecomment-2058922899.

## Problem

While reviewing woocommerce/woocommerce-blocks#8819, I saw that we're currently showing notices in the shipping section on the Checkout block like this:

<img width="851" alt="Screenshot 2023-03-28 at 13 02 00" src="https://user-images.githubusercontent.com/3323310/228142837-4700cd3b-dcc6-4c7f-95f1-15ba0262a03b.png">

In woocommerce/woocommerce-blocks#8659, @mikejolley is working on new components and styling for buyer notices. Once that PR has been merged, we should address the buyer's notice of shipping section within Checkout block.

**Comment by ralucaStan** [comment]
Nice catch Niels. 

The idea is that we show a notice in the Cart Order summary as seen below

<img width="640" alt="Screenshot 2023-04-10 at 13 58 04" src="https://user-images.githubusercontent.com/1628454/230889752-4ad0a2bc-5b11-4c10-804e-ed6a448496d8.png">

And we show a notice in the Checkout block > Shipping section. 
<img width="384" alt="Screenshot 2023-04-10 at 14 01 34" src="https://user-images.githubusercontent.com/1628454/230889657-26a4f3de-4705-4abd-a02b-1586782a3b2e.png">

@pmcpinto @elizaan36 does it make sense to show another notice in the right side of the Checkout block as well?

**Comment by ralucaStan** [comment]
Worth mentioning that we should also look at the clean state for both blocks when no address is entered

<img width="855" alt="Screenshot 2023-04-10 at 14 26 32" src="https://user-images.githubusercontent.com/1628454/230893099-5dd1a19e-020e-430e-b46b-63710e07b531.png">
<img width="828" alt="Screenshot 2023-04-10 at 14 26 23" src="https://user-images.githubusercontent.com/1628454/230893118-5910e7c1-ea49-4039-af48-6a05d2fb0c83.png">

My proposal is to align the states of the Order summary shipping section in all cases:

1. Empty addresses
Cart block: Show link to add "Add an address for shipping options" that shows the Shipping calculator text
Checkout block: Add the same text, maybe in a notice? @elizaan36 

2. Address outside of shipping areas:
Cart block: keep as is: show the address and warning notice 
Checkout block: align with the Cart and show the address and warning notice

**Comment by pmcpinto** [comment]
Thanks for the ping. Here's my feedback:

> Cart block: Show link to add "Add an address for shipping options" that shows the Shipping calculator text

Sounds good to me.

> Checkout block: Add the same text, maybe in a notice? @elizaan36

+1 on using the same text, but I think the notice can be distracting

> Cart block: keep as is: show the address and warning notice

+1

> Checkout block: align with the Cart and show the address and warning notice

I'm afraid that the notice may be redundant. 

Let's wait for Elizabeth's feedback before making any changes. Thanks

**Comment by elizaan36** [comment]
> @pmcpinto @elizaan36 does it make sense to show another notice in the right side of the Checkout block as well?

+1 on [Pedro's notes here](https://github.com/woocommerce/woocommerce/issues/42422). We should only show the notice once on the page in case of issues with the address, in the Cart block order summary and the Shipping options on the checkout block.

In the clean slate view for the checkout block, I think the plain text "There are no shipping methods available" is sufficient. The notice isn't needed for the NUX flow, and if there's any issue with the address the notice will appear in the Shipping options section.

**Comment by nielslange** [comment]
@elizaan36 In https://github.com/woocommerce/woocommerce/issues/42422, you mentioned

> _"We should only show the notice once on the page in case of issues with the address, in the Cart block order summary and the Shipping options on the checkout block"._

What are your thoughts regarding the Checkout block's shipping section? Should we display only the shipping label or hide the entire section? Below, I've attached screenshots of both options.

<table>
<tr>
<td valign="top">Empty shipping section:
<br><br>
<img width="483" alt="Screenshot 2023-11-24 at 19 25 51" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/1ca379b9-1a6c-4cdc-b993-5e1d713488e7">
</td>
<td valign="top">Hidden shipping section:
<br><br>
<img width="479" alt="Screenshot 2023-11-24 at 19 26 12" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/a20400d0-4ca0-458d-861f-76dbd4a6cb97">
</td>
</tr>
</table>

**Comment by ralucaStan** [comment]
I suggest keeping these sections and showing `0` or `-` for the values.

**Comment by nielslange** [comment]
> I suggest keeping these sections and showing `0` or `-` for the values.

If we keep that section, I'd rather show `-` as `0` might imply that shipping is free. What are your thoughts on that, @ralucaStan?

**Comment by ralucaStan** [comment]
Sounds good to me.

**Comment by nielslange** [comment]
@ralucaStan While working on this issue and checking edge cases, I remembered the issue woocommerce/woocommerce#42124. In that issue, the reported mentioned that it's unexpected to see `No shipping option available` when no shipping address has been provided, but shipping options are available in general. I've drafted https://github.com/woocommerce/woocommerce-blocks/pull/11983 to demonstrate how both this issue and woocommerce/woocommerce#42124 could be resolved. 

@elizaan36 and @pmcpinto Based on https://github.com/woocommerce/woocommerce/issues/42422 and https://github.com/woocommerce/woocommerce/issues/42422, I understand that you want to avoid redundant notices. On a desktop, both affected sections are visible. However, on a mobile device, only one of the sections would be visible. The order summary section is the first section a buyer would see. To increase visibility and to avoid that a buyer has to scroll down to find important information, I was wondering if it wouldn't be better to show the notice in both sections, despite being redundant. The following screenshots show the message in the order summary section as plain text vs. as a warning notice. I've also added a screenshot to show how far below the fold the shipping warning notice is when using a mobile device (in my case, iPhone SE with a resolution of 375 x 667).

<table>
<tr>
<td valign="top">Plain text:
<br><br>
<img width="397" alt="Screenshot 2023-11-29 at 18 13 11" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/c6dd4850-22ce-40a4-b393-2d5be5074c43">
</td>
<td valign="top">Warning notice:
<br><br>
<img width="393" alt="Screenshot 2023-11-29 at 18 16 17" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/d7730a01-2d1d-4744-ba56-4526e746d227">
</td>
<td valign="top">Entire mobile page:
<br><br>
<img width="251" alt="Screenshot 2023-11-29 at 18 17 38" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/95cd23f1-c5d7-4ec1-b64f-e5c8b635b4f6">
</td>
</tr>
</table>

**Comment by sunyatasattva** [comment]
Just jumping in with my unsolicited opinion 😬 

I've gotta say that I thoroughly agree with @nielslange on this issue. From a UX standpoint, I think it should be made entirely clear that the shipping costs are going to appear once the address has been filled up, as opposed to showing either a vague `-` or—worse—the “No shipping options available” (as pointed out by the merchant in woocommerce/woocommerce#42124).

Even moreso, considering that the blocks can appear in any order and in any configuration we can't predict; so, while it may be redundant, I think in this case such redundancy is justifiable or even warranted. On the actual design of the text I have no strong opinion, as I understand we might want to avoid too many notices making things busy or alarming for users. The text might be a little italics label, or anything else: perhaps adding some other notice style (off the top of my head I can imagine a background-less notice with just the icon and maybe a faint border).

But, regardless, I think we should be as clear as possible to the user, or otherwise remove the section altogether as @nielslange [suggested above](https://github.com/woocommerce/woocommerce/issues/42422) in case of empty address. Still, my preference would be for what Niels is proposing in woocommerce/woocommerce-blocks#11983.

**Comment by ralucaStan** [comment]
Thank you @nielslange for your care for the overall experience when exploring solutions for this ticket.

I took a deep dive into the topic and I've noticed that there is a WooCommerce setting that is not respected by the  Checkout & Cart blocks:

<img width="559" alt="Screenshot 2023-11-29 at 15 35 03" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/a5572aaa-a07d-4d33-99c8-47042ea03fbe">


When **this setting is active**, the shortcodes hide the Shipping entry from the totals Listing (both Cart and Checkout).

When the **setting is not active**, the shortcodes show these messages:

| Cart block > No shipping calculator  | Cart block > With Shipping calculator |
| ------------- | ------------- |
| <img width="610" alt="Screenshot 2023-11-29 at 15 40 22" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/3a627769-7d65-4cb1-8ecf-3766f3012725">  | <img width="814" alt="Screenshot 2023-11-29 at 15 41 20" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/d25b9968-89d8-4978-91c2-c0e74c26f56c">  |

| Checkout block  |
| ------------- |
| <img width="816" alt="Screenshot 2023-11-29 at 15 40 44" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/b83bb363-e40c-4c12-a598-2341118772a0">  |

---
**My proposal**: we should also support the WooCommerce setting; which eliminates the Shipping area from the totals. That will just solve half of the problem....

## Zooming out
<img width="834" alt="Screenshot 2023-11-29 at 15 47 15" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/ccb6e547-cf71-45e1-b997-ba0576816ca4">

The Checkout shortcode does not have a Shipping options section. The options are directly displayed in the Order summary. 
That's why the shortcode display  `Enter your address to view shipping options` (costs are near shipping options).

These messages were ported to the **Checkout block order summary** as follows:
`No shipping options available`, or as you changed it @nielslange , to `Add an address for shipping options`. 

The **Checkout block order summary** does not display options, only costs, and this is what led to user confusion in https://github.com/woocommerce/woocommerce/issues/42124

### Shipping related prompts in the checkout block

This is what is currently displayed. Notice that we don't show notices, as those are only shown when there are no options for the entered address.


| Totals in Order summary  | Shipping options section |
| ------------- | ------------- |
| <img width="277" alt="Screenshot 2023-11-29 at 16 00 28" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/c66924c8-06b6-41a7-af73-fb1405195860">  | <img width="522" alt="Screenshot 2023-11-29 at 16 00 24" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/47198ccb-3ddc-4d5c-96d6-984fc1caccaf">  |

----

**My proposal**: Align the message tone in the Order summary and mention Shipping costs instead of options: `Select a shipping option to see cost/price. Pros: let's the user know what to do, and what will be displayed as a result.
 I'll let the decision around how we display this message to @elizaan36 . 

Going forward, I'd also align the message we show in the Shipping options section with `Enter your address to view shipping options` instead of `Add a shipping address to view shipping options.`. 
- It's shorter
- It aligns with the shortcode version, and it's easier to understand what scenario the text addresses in both scenarios

**Comment by ralucaStan** [comment]
> I'll let the decision around how we display this message to @elizaan36 .

My preference is to keep the text, but discard the italic and bold styling. This would work on all devices.

**Comment by nielslange** [comment]
Thanks for your reply, @ralucaStan.

> My proposal: we should also support the WooCommerce setting; which eliminates the Shipping area from the totals. That will just solve half of the problem....

I agree that we should reflect the setting `Hide shipping costs until an address is entered` within the Cart and Checkout blocks.

---

> These messages were ported to the **Checkout block order summary** as follows:
`No shipping options available`, or as you changed it @nielslange, to `Add an address for shipping options`.

In woocommerce/woocommerce-blocks#11983, I've added the following conditional logic for displaying these messages:
- When no address has been provided, show `Add an address for shipping options`.
- When an address has been provided, but no shipping option is available for this location, show `No shipping options available`.

---

> The Checkout block order summary does not display options, only costs, and this is what led to user confusion in https://github.com/woocommerce/woocommerce/issues/42124

My understanding of woocommerce/woocommerce#42124 is, that the confusion is caused because we're showing `No shipping options available` as a point where a buyer expects to see `Add an address for shipping options`. IMO, we should only show `No shipping options available` once we're certain that there's no shipping option for the provided address. 

---

> This is what is currently displayed. Notice that we don't show notices, as those are only shown when there are no options for the entered address.

That's correct, but looking at the following scenarios, the mix of notice and plain text looks inconsistent to me:

### Scenario 1

<img width="377" alt="Screenshot 2023-11-30 at 18 19 25" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/50dfb085-407f-47bc-b222-c666f72bbd3c">

<img width="771" alt="Screenshot 2023-11-30 at 18 19 45" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/27a2a6e4-bba0-4500-b186-8840106e9e2f">

### Scenario 2

<img width="378" alt="Screenshot 2023-11-30 at 18 20 40" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/62272501-0f9a-4b24-a831-4e316e1d3efa">

<img width="505" alt="Screenshot 2023-11-30 at 18 20 54" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/082f37fd-9c33-4011-a807-53bc577d1257">

### Scenario 3

<img width="373" alt="Screenshot 2023-11-30 at 18 22 11" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/aed44a8c-eae0-4e2e-ba83-14683cc7a71f">

<img width="776" alt="Screenshot 2023-11-30 at 18 21 14" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/2c25ebc2-f10d-4fcd-81b5-ba1e4b937147">

### Scenario 4

<img width="375" alt="Screenshot 2023-11-30 at 18 23 21" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/393683d3-fcf2-4bbd-8a21-30dfb827ab7a">

<img width="511" alt="Screenshot 2023-11-30 at 18 22 54" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/e4050724-5518-4fa3-8198-35c9e890d137">

---

> **My proposal**: Align the message tone in the Order summary and mention Shipping costs instead of options: `Select a shipping option to see cost/price. Pros: let's the user know what to do, and what will be displayed as a result.

Can you clarify this comment? My understanding of this issue is that it addresses the case when no shipping options are available. Either because the buyer hasn't provided an address yet, or there are no shipping options available for the provided address. If the buyer provided an address for which a shipping option exists, we do show the shipping costs:

<img width="1242" alt="Screenshot 2023-11-30 at 18 31 03" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/30c8c317-83dd-434a-b953-85ff7f412eb4">

---

> Going forward, I'd also align the message we show in the Shipping options section with `Enter your address to view shipping options` instead of `Add a shipping address to view shipping options.`.

I agree that we should align them. Apart from that, we might also want to look at all messages and align them:

- `Add an address for shipping options` → Checkout block / Order summary
- `Add a shipping address to view shipping options.` → Checkout block / Shipping options
- `Calculated during checkout` → Cart block
- `Calculate shipping` → Cart shortcode
- `Change address` → Cart block
- `Enter your address to view shipping options.` → Checkout shortcode
- `No shipping options available` → Checkout block / Order summary
- `No shipping options were found for {ADDRESS}` → Cart shortcode
- `Shipping costs are calculated during checkout.` → Cart shortcode
- `Shipping options will be displayed here after entering your full shipping address.` → Checkout block / Shipping options
- `There are no shipping options available. Please check your shipping address.` → Checkout block / Shipping options
- `There are no shipping options available. Please ensure that your address has been entered correctly, or contact us if you need any help.` → Checkout shortcode

**Comment by nielslange** [comment]
@ralucaStan Based on https://github.com/woocommerce/woocommerce/issues/42422, I've investigated how the `Enable the shipping calculator on the cart page` and `Hide shipping costs until an address is entered` settings affect the shortcode-based and block-based checkout flow. 

## Testing options

I've tested all possibilities for the following options:
- Enable the shipping calculator on the cart page: Checked / Unchecked
- Hide shipping costs until an address is entered: Checked / Unchecked
- Address: Not provided / Provided, but no shipping option available

## Testing sections

During the tests, I kept an eye on the following sections:
- Cart shortcode (Order Summary)
- Cart block (Order Summary)
- Checkout shortcode (Order Summary)
- Checkout block (Order Summary)
- Checkout block (Shipping Options)

## Test scenarios

Based on the listed testing options and testing sections, I listed above, I created the following test scenarios. For each scenario, I've mentioned the conditions, the texts and notices that are currently visible, and added screenshots.

<details>
<summary>Scenario 1</summary>

<br>

- Enable the shipping calculator on the cart page: **Unchecked**
- Hide shipping costs until an address is entered: **Unchecked**
- Address: **Not provided**

<br>

<table>
    <tr>
        <th valign="top">Cart shortcode<br>(Order Summary)</th>
        <th valign="top">Cart block<br>(Order Summary)</th>
        <th valign="top">Checkout shortcode<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Shipping Options)</th>
    </tr>
    <tr>
        <td valign="top">Show text “Shipping costs are calculated during checkout.”</td>
        <td valign="top">Show text “Calculated during checkout”</td>
        <td valign="top">Show text “Enter your address to view shipping options.”</td>
        <td valign="top">Show text “Add an address for shipping options”</td>
        <td valign="top">Show text “Add a shipping address to view shipping options.”</td>
    </tr>
</table>

## Cart

<img width="2880" alt="Screenshot 2023-11-30 at 15 52 30" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/1eaf3fbf-44d3-4c27-91dc-c8f736cec2ba">

## Checkout

<img width="2880" alt="Screenshot 2023-11-30 at 15 54 40" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/e95415f7-f237-4d3a-80ab-6e947e6ccd3a">

</details>

---

<details>
<summary>Scenario 2</summary>

<br>

- Enable the shipping calculator on the cart page: **Checked**
- Hide shipping costs until an address is entered: **Unchecked**
- Address: **Not provided**

<br>

<table>
    <tr>
        <th valign="top">Cart shortcode<br>(Order Summary)</th>
        <th valign="top">Cart block<br>(Order Summary)</th>
        <th valign="top">Checkout shortcode<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Shipping Options)</th>
    </tr>
    <tr>
        <td valign="top">Show text “Enter your address to view shipping options.” and link “Calculate shipping”</td>
        <td valign="top">Show link “Add an address for shipping options"</td>
        <td valign="top">Show text “Enter your address to view shipping options.”</td>
        <td valign="top">Show text “Add an address for shipping options”</td>
        <td valign="top">Show text “Add a shipping address to view shipping options.”</td>
    </tr>
</table>

## Cart

<img width="2880" alt="Screenshot 2023-11-30 at 15 58 35" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/91d3039d-25ae-4e84-bf24-0d3fb881a612">

## Checkout

<img width="2880" alt="Screenshot 2023-11-30 at 16 00 35" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/f37f2e6b-c22c-4cc1-824a-9f7a073f2dfd">

</details>

---

<details>
<summary>Scenario 3</summary>

<br>

- Enable the shipping calculator on the cart page: **Unchecked**
- Hide shipping costs until an address is entered: **Checked**
- Address: **Not provided**

<br>

<table>
    <tr>
        <th valign="top">Cart shortcode<br>(Order Summary)</th>
        <th valign="top">Cart block<br>(Order Summary)</th>
        <th valign="top">Checkout shortcode<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Shipping Options)</th>
    </tr>
    <tr>
        <td valign="top">No section visible</td>
        <td valign="top">Show text “Calculated during checkout”</td>
        <td valign="top">No section visible</td>
        <td valign="top">Show text “Add an address for shipping options”</td>
        <td valign="top">Show warning notice “Shipping options will be displayed here after entering your full shipping address.”</td>
    </tr>
</table>

## Cart

<img width="2880" alt="Screenshot 2023-11-30 at 16 03 01" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/36e44e32-baa1-4d69-976b-b542d9dc45b2">

## Checkout

<img width="2880" alt="Screenshot 2023-11-30 at 16 06 10" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/0b648e73-4728-4213-ac7c-fff7996d5bdc">

</details>

---

<details>
<summary>Scenario 4</summary>

<br>

- Enable the shipping calculator on the cart page: **Checked**
- Hide shipping costs until an address is entered: **Checked**
- Address: **Not provided**

<br>

<table>
    <tr>
        <th valign="top">Cart shortcode<br>(Order Summary)</th>
        <th valign="top">Cart block<br>(Order Summary)</th>
        <th valign="top">Checkout shortcode<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Shipping Options)</th>
    </tr>
    <tr>
        <td valign="top">Show link “Calculate shipping”</td>
        <td valign="top">Show link “Add an address for shipping options”</td>
        <td valign="top">No section visible</td>
        <td valign="top">Show text “Add an address for shipping options”</td>
        <td valign="top">Show warning notice “Shipping options will be displayed here after entering your full shipping address.”</td>
    </tr>
</table>

## Cart

<img width="2880" alt="Screenshot 2023-11-30 at 16 13 13" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/4eae72da-dde6-4c0b-b9b8-5bd46e03c2b8">

## Checkout

<img width="2880" alt="Screenshot 2023-11-30 at 16 14 54" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/a16ef611-24ed-4ca7-9571-28973e906c99">

</details>

---

<details>
<summary>Scenario 5</summary>

<br>

- Enable the shipping calculator on the cart page: **Unchecked**
- Hide shipping costs until an address is entered: **Unchecked**
- Address: **Provided, but no shipping option available**

<br>

<table>
    <tr>
        <th valign="top">Cart shortcode<br>(Order Summary)</th>
        <th valign="top">Cart block<br>(Order Summary)</th>
        <th valign="top">Checkout shortcode<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Shipping Options)</th>
    </tr>
    <tr>
        <td valign="top">Show text “No shipping options were found for {ADDRESS}" and link “Enter a different address”</td>
        <td valign="top">Show {ADDRESS}, link “Change address” and warning notice “There are no shipping options available. Please check your shipping address.”</td>
        <td valign="top">Show text “There are no shipping options available. Please ensure that your address has been entered correctly, or contact us if you need any help.”</td>
        <td valign="top">Show text “No shipping options available”</td>
        <td valign="top">Show text “There are no shipping options available. Please check your shipping address.”</td>
    </tr>
</table>

## Cart

<img width="2880" alt="Screenshot 2023-11-30 at 16 30 32" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/6dc6d78e-edfd-4586-8f6e-302668103c67">

## Checkout

<img width="2880" alt="Screenshot 2023-11-30 at 16 43 01" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/a6f34a74-1884-47a6-9131-1d9df2ec40e0">

</details>

---

<details>
<summary>Scenario 6</summary>

<br>

- Enable the shipping calculator on the cart page: **Checked**
- Hide shipping costs until an address is entered: **Unchecked**
- Address: **Provided, but no shipping option available**

<br>

<table>
    <tr>
        <th valign="top">Cart shortcode<br>(Order Summary)</th>
        <th valign="top">Cart block<br>(Order Summary)</th>
        <th valign="top">Checkout shortcode<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Shipping Options)</th>
    </tr>
    <tr>
        <td valign="top">Show text “No shipping options were found for {ADDRESS}” and link “Enter a different address”</td>
        <td valign="top">Show {ADDRESS}, link “Change address” and warning notice “There are no shipping options available. Please check your shipping address.”</td>
        <td valign="top">Show text “There are no shipping options available. Please ensure that your address has been entered correctly, or contact us if you need any help.”</td>
        <td valign="top">Show text “No shipping options available”</td>
        <td valign="top">Show warning notice “There are no shipping options available. Please check your shipping address.”</td>
    </tr>
</table>

## Cart

<img width="2880" alt="Screenshot 2023-11-30 at 17 00 43" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/7f9e27f1-47c2-4932-96e2-dcf3edddcbbd">

## Checkout

<img width="2880" alt="Screenshot 2023-11-30 at 17 02 35" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/44ffcd72-c22e-49e6-9657-17f4c14587a7">

</details>

---

<details>
<summary>Scenario 7</summary>

<br>

- Enable the shipping calculator on the cart page: **Unchecked**
- Hide shipping costs until an address is entered: **Checked**
- Address: **Provided, but no shipping option available**

<br>

<table>
    <tr>
        <th valign="top">Cart shortcode<br>(Order Summary)</th>
        <th valign="top">Cart block<br>(Order Summary)</th>
        <th valign="top">Checkout shortcode<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Shipping Options)</th>
    </tr>
    <tr>
        <td valign="top">Show text “No shipping options were found for {ADDRESS}”</td>
        <td valign="top">Show {ADDRESS} and warning notice “There are no shipping options available. Please check your shipping address.”</td>
        <td valign="top">Show text “There are no shipping options available. Please ensure that your address has been entered correctly, or contact us if you need any help.”</td>
        <td valign="top">Show text “No shipping options available”</td>
        <td valign="top">Show warning notice “There are no shipping options available. Please check your shipping address.”</td>
    </tr>
</table>

## Cart

<img width="2880" alt="Screenshot 2023-11-30 at 17 10 15" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/21c2418b-d6c5-4e84-a899-13c5af137824">

## Checkout

<img width="2880" alt="Screenshot 2023-11-30 at 17 11 52" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/8712d155-207d-49b5-ba93-a9245a107427">

</details>

---

<details>
<summary>Scenario 8</summary>

<br>

- Enable the shipping calculator on the cart page: **Checked**
- Hide shipping costs until an address is entered: **Checked**
- Address: **Provided, but no shipping option available**

<br>

<table>
    <tr>
        <th valign="top">Cart shortcode<br>(Order Summary)</th>
        <th valign="top">Cart block<br>(Order Summary)</th>
        <th valign="top">Checkout shortcode<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Order Summary)</th>
        <th valign="top">Checkout block<br>(Shipping Options)</th>
    </tr>
    <tr>
        <td valign="top">Show text “No shipping options were found for {ADDRESS}” and link “Enter a different address”</td>
        <td valign="top">Show {ADDRESS} and warning notice “There are no shipping options available. Please check your shipping address.”</td>
        <td valign="top">Show text “There are no shipping options available. Please ensure that your address has been entered correctly, or contact us if you need any help.”</td>
        <td valign="top">Show text “No shipping options available”</td>
        <td valign="top">Show warning notice “There are no shipping options available. Please check your shipping address.”</td>
    </tr>
</table>

## Cart

<img width="2880" alt="Screenshot 2023-11-30 at 17 56 24" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/83b038bc-338c-4157-b86f-3146e6b5e18d">

## Checkout

<img width="2880" alt="Screenshot 2023-11-30 at 17 57 51" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/43bdc1e6-ac31-46e4-96be-eeaf9580dbff">

</details>

## Proposal

I propose individually reviewing each testing scenario to determine what to display for each case. As for the use of notices instead of plain texts, pdFofs-11n-p2 states that they should be used _"To display a concise, important message that requires the user’s attention."_ IMO, the examples listed, in which we show plain text messages, e.g. _"Add an address for shipping options"_, as mentioned in Scenario 1, qualify for this.

cc: @elizaan36 and @pmcpinto

**Comment by elizaan36** [comment]
@nielslange Sorry for the delay on this one, I only just returned to this after providing feedback to Tarun on a [similar issue](https://github.com/woocommerce/woocommerce/issues/42280#issuecomment-1900944615).

> What are your thoughts regarding the Checkout block's shipping section? Should we display only the shipping label or hide the entire section?

I don't think we should hide the line item from the order summary if the address hasn't been entered, however the current helper text is repetitive with the shipping options copy (as you've shown in Scenarios 1 and 2.) We could shorten the text in the order summary to "Enter shipping address".

In response to @ralucaStan 's comment:
> Going forward, I'd also align the message we show in the Shipping options section with "Enter your address to view shipping options" instead of Add a shipping address to view shipping options.

Makes sense to me. We can make that holistic change on the Cart and Checkout. cc @nikkivias for your thoughts here. I'll dive into the additional scenarios in a new comment.

**Comment by elizaan36** [comment]
> Scenario 3 - Checkout
> Scenario 4 - Checkout

This warning notice style in Shipping Options isn't correct. It should be a simple text string with the copy "Enter your address to view shipping options". I'll share a screenshot of @nikkivias ' recent design to illustrate. Nikki, feel free to share your feedback on this issue as well. :)

<img width="1431" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5121465/1d7c1c88-f013-4e16-b615-57cade1d0544">

The Cart seems fine in these scenarios barring any copy and styling changes.

> Scenarios 5-8 (Cart & Checkout)

In these use cases where the address has been entered and there are no shipping options, the warning notice is correct. 

Regarding the styling comment from Raluca,
> My preference is to keep the text, but discard the italic and bold styling. This would work on all devices.

+1. The bold, and especially the bold + italic is unnecessary. I see that Nikki has updated the styles in her latest designs.

**Comment by pmcpinto** [comment]
@ralucaStan @elizaan36 @nikkivias we have some additional issues in our backlog that touch this area:
#45525
#42124 
#46014

Improving the shopper experience is a current priority. Therefore, this is the right time to review all the use cases, consolidate everything into one issue, and try to implement the changes in cycle 3. LMK what you think.

**Comment by ralucaStan** [comment]
Nice catch @pmcpinto! I think it makes sense to align the experience now that we have a design solution in place.

Just noting that https://github.com/woocommerce/woocommerce/issues/45525 and https://github.com/woocommerce/woocommerce/issues/42124 still need to be updated with the expected result, making sure that this will be the case before we start the Cycle.

What are your thoughts on updating those tickets Pedro?

**Comment by dominique-vias** [comment]
@ralucaStan  @pmcpinto The way I see it there should be appropriate notices in stages. First entry state (no address entered yet), and no shipping options available based on address input. Is my understanding correct? Are there other scenarios we must account for?

I also would like to clarify something— if shipping zones are limited to certain areas, they shouldn't have been able to enter their full address right? For example, country / city / region dropdown selection would have been limited?

**Comment by ralucaStan** [comment]
@nikkivias could you share some visual support or testing steps? That would help me understand better your questions. Thank you so much

**Comment by dominique-vias** [comment]
Hi @ralucaStan here is a visual! I'm mainly wondering if this error scenario possible?

![CleanShot 2024-04-11 at 16 52 13@2x](https://github.com/woocommerce/woocommerce/assets/99638528/3585c76b-e637-48c8-a811-3880c5798beb)

My thinking is this error wouldn't exist if supported shipping zones are reflected in the dropdown selections. For example, if I only ship to US and UK, I can only select US and UK in the dropdown.

Or is it still possible for the error to occur even if they selected an address that is in the supported zones?

![CleanShot 2024-04-11 at 17 06 45@2x](https://github.com/woocommerce/woocommerce/assets/99638528/8d226f4e-1bdf-4b3a-a09d-22ce8c011ff3)

**Comment by pmcpinto** [comment]
> I also would like to clarify something— if shipping zones are limited to certain areas, they shouldn't have been able to enter their full address right? For example, country / city / region dropdown selection would have been limited?

@nikkivias when a store sells only to specific countries, we only show the selected countries in the dropdown. IMO the experience is a bit confusing because, by default, we don't mention anywhere that the shipping is limited to X,Y,Z country. When you try to search for a country, and it doesn't show the country you're looking for, it looks more like a bug than a feature. But I'm conscious that eventually there's no straightforward way to provide this context, because the merchant can cherry-pick 1, 10, 100... countries.

**Comment by pmcpinto** [comment]
> What are your thoughts on updating those tickets Pedro?

@ralucaStan I think the best option is to create a new issue with the solution(s) and close all the existing issues. We can point them to the new issue. Does that make sense?

**Comment by pmcpinto** [comment]
> @ralucaStan @pmcpinto The way I see it there should be appropriate notices in stages. First entry state (no address entered yet), and no shipping options available based on address input. Is my understanding correct? Are there other scenarios we must account for?

@nikkivias we have the following surface areas:

Cart - cart totals - shipping section
Checkout - checkout fields - shipping section
Checkout - checkout totals - shipping section

All the scenarios are summarized in this excellent analysis made by Niels: https://github.com/woocommerce/woocommerce/issues/42422#issuecomment-1850746031

**Comment by ralucaStan** [comment]
It's also not just about the country, a shopper can match on 3 shipping zones https://woocommerce.com/document/setting-up-shipping-zones/#section-3 zip code, country and continent based.

**Comment by ralucaStan** [comment]
> Hi @ralucaStan here is a visual! I'm mainly wondering if this error scenario possible?

I replicated this on my testing site. This is what I see when entering an address with no shipping available. 

![Image](https://github.com/woocommerce/woocommerce/assets/1628454/5cd778df-34bb-4b48-85dd-adc33faccf98)

This was only visible when I entered the full address, for a country that was not supported in the shipping options. Before the notice appeared I saw this 



![Image](https://github.com/woocommerce/woocommerce/assets/1628454/aeed7fd7-7c7c-4c9c-a504-ca25dfb958bb)

**Comment by ralucaStan** [comment]
I'm marking this as blocked, as we are still defining the outcome

**Comment by pmcpinto** [comment]
@nikkivias when do you think you'll have time to dive into this area?

**Comment by dominique-vias** [comment]
I am working on this now.

**Comment by dominique-vias** [comment]
Hi team @ralucaStan @pmcpinto  I created screens based on the scenarios outlined above. Thanks @nielslange they were super helpful. please check this [design file](https://www.figma.com/file/c8VYjXqluE9vlVSE1RgJix/WooCommerce-Cart-%26-Checkout?type=design&node-id=0%3A1&mode=design&t=FhucG0I92QzEULpf-1). As part of our experimental initiative to default to public, I made this design file viewable to the public.

### Some high-level changes:

- I made sure that messages are aligned as much as possible in both Cart & checkout, as well as short-code and blocks, so that we are not using too many variations.

- Changed the word `Shipping` to `Delivery` where appropriate. Pickup is offered as an option so `Shipping` as a row label, and `shipping options` are no longer appropriate. Since we are making these changes in the checkout fields, it makes sense to update Summary labels too
    - What I am not sure about is whether this change should only apply if Pickup is enabled, or should we finally decide to make this a holistic change? 🤔
    
- We don't use any notice components during the `Cart` stage

### Other

As I was going through the scenarios, it seems to me that this setting:

`Hide shipping costs until an address is entered`

almost seems pointless because it is expected behavior for shipping options / costs to not show until an address is entered. Or am I missing something?

**Comment by nielslange** [comment]
Thanks for creating the designs, @nikkivias, and I'm glad to hear that the scenarios, I've listed in https://github.com/woocommerce/woocommerce/issues/42422#issuecomment-1850746031 were helpful.

As for `Hide shipping costs until an address is entered`, if I understand you correctly, you are proposing of removing that option entirely, correct? I'm asking as on Baymard, I found the following two guidelines:
- [Present All Shipping Options Directly on the Page](https://baymard.com/premium/guideline-collections/nc5h8z/547?businessObjective=mpuuww)
- [Always Integrate “Free Shipping” Information in the Cart Order Cost Summary](https://baymard.com/premium/guideline-collections/40rcjv/2320)

**Comment by dominique-vias** [comment]
> As for `Hide shipping costs until an address is entered`, if I understand you correctly, you are proposing of removing that option entirely, correct?

Yes. As I was going through the scenarios you outlined, I didn't really see the difference in how the setting presents itself whether it is checked or unchecked. Am I seeing that right?

✅ Calc ✅ Hide costs = meaning hide costs if no address entered. If address is provided, show costs.

Isn't that already the expected behavior? Even if `Hide costs` were unchecked, there will be no cost to show until an address is entered. Sorry if I may not be explaining it well.

**Comment by nielslange** [comment]
> Yes. As I was going through the scenarios you outlined, I didn't really see the difference in how the setting presents itself whether it is checked or unchecked. Am I seeing that right?

@nikkivias The changes are minimal and easy to oversee. This is what I see on my end:

<table>
    <thead>
        <tr>
            <th align="left" valign="top" width="10%">Enable the shipping calculator on the cart page</th>
            <th align="left" valign="top" width="10%">Hide shipping costs until an address is entered</th>
            <th align="left" valign="top" width="20%">Classic cart</th>
            <th align="left" valign="top" width="20%">Cart block</th>
            <th align="left" valign="top" width="20%">Classic checkout</th>
            <th align="left" valign="top" width="20%">Checkout block</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td align="left" valign="top">☐ unchecked</td>
            <td align="left" valign="top">☐ unchecked</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are visible <br><br> No text displayed</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are visible <br><br> No text displayed</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are visible <br><br> No text displayed</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping method is visible <br><br> No text displayed</td>
        </tr>
        <tr>
            <td align="left" valign="top">☑ checked</td>
            <td align="left" valign="top">☐ unchecked</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are visible <br><br> Text "Calculate shipping" is displayed</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are visible <br><br> No text displayed</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are visible <br><br> No text displayed</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping method is visible <br><br> No text displayed</td>
        </tr>
        <tr>
            <td align="left" valign="top">☐ unchecked</td>
            <td align="left" valign="top">☑ checked</td>
            <td align="left" valign="top">Shipping section is hidden <br><br> Shipping methods are hidden <br><br> No text displayed</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are hidden <br><br> Text "Calculated during checkout" is displayed</td>
            <td align="left" valign="top">Shipping section is hidden <br><br> Shipping methods are hidden <br><br> No text displayed</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are hidden <br><br> Text "No shipping options available" is displayed</td>
        </tr>
        <tr>
            <td align="left" valign="top">☑ checked</td>
            <td align="left" valign="top">☑ checked</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are hidden <br><br> Text "Calculate shipping" is displayed</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are hidden <br><br> Text "Add an address for shipping options" is displayed</td>
            <td align="left" valign="top">Shipping section is hidden <br><br> Shipping methods are hidden <br><br> No text displayed</td>
            <td align="left" valign="top">Shipping section is visible <br><br> Shipping methods are hidden <br><br> Text "No shipping options available" is displayed</td>
        </tr>
    </tbody>
</table>

When `Hide shipping costs until an address is entered` is enabled, we show a text on the Cart block, which we do don't show when the that option is disabled. When `Enable the shipping calculator on the cart page` is enabled, we show the text `Add an address for shipping options`. Otherwise, we show the text `Calculated during checkout`. As you can see, the changes are minimal.

**Comment by pmcpinto** [comment]
> Hi team @ralucaStan @pmcpinto I created screens based on the scenarios outlined above. Thanks @nielslange they were super helpful. please check this c8VYjXqluE9vlVSE1RgJix-fi-0%3A1. As part of our experimental initiative to default to public, I made this design file viewable to the public.

@nikkivias thanks for sharing the file. Looks good to me. I've just added comments about some details.

> Hide shipping costs until an address is entered
> almost seems pointless because it is expected behavior for shipping options / costs to not show until an address is entered. Or am I missing something?

Yes. The differences are almost undetectable. But I think we should keep it for now, to avoid breaking any kind of backwards compact.

**Comment by pmcpinto** [comment]
@nikkivias is wrapping up the mockups covering all the use cases.

@ralucaStan should we use this issue as the source of truth for all the shipping notices changes or create a new one?

**Comment by ralucaStan** [comment]
I think we should start fresh; this issue has a lot of discussions and we can tag it in the new issue; @nikkivias, @nielslange who wants to create the new issue using the new [design file](https://www.figma.com/file/c8VYjXqluE9vlVSE1RgJix/WooCommerce-Cart-%26-Checkout?type=design&node-id=0%3A1&mode=design&t=FhucG0I92QzEULpf-1)

**Comment by nielslange** [comment]
I can take care of it, @ralucaStan. I'd actually create one main issue and, for each case, a separate sub issue. That way, it's easier to implement, test and revert (in case something goes wrong).

**Comment by nielslange** [comment]
Closing this issue in benefit of #46945, see https://github.com/woocommerce/woocommerce/issues/42422#issuecomment-2076697149 for the reasoning.
